docs/root/configuration/http_conn_man/headers.rst

```diff
@@ -299,3 +299,12 @@ Supported dynamic values are:
 %PROTOCOL%
     The original protocol which is already added by Envoy as a
     :ref:`x-forwarded-proto <config_http_conn_man_headers_x-forwarded-proto>` request header.
+
+%UPSTREAM_METADATA(namespace, key[, key...])%
```

Comment:
I kind of worry about parsing here being a big ugly when it comes to escaping spaces, quotes, dots etc. Do you have any thoughts on how we can specify this to allow cheap parsing? E.g. if the key list is a JSON list, we could reuse the JSON parser, etc.

Reply:
Maybe something like:

```%UPSTREAM_METADATA("namespace", "foo", "bar")%```

and then document that the strings are escaped as in JSON. And in envoy stuff some brackets around it and use the JSON parser?

Reply:
What about `%UPSTREAM_METADATA(["namespace", "foo", "bar"])%` ? Just actual JSON?

Reply:
Sure. Probably get better error messages that way.